lib/eslint/eslint-helpers.js

```diff
@@ -569,17 +602,17 @@ async function findFiles({
         // save patterns for later use based on whether globs are enabled
         if (globInputPaths && isGlobPattern(pattern)) {
 
+            /*
+             * We are grouping patterns by their glob parent. This is done to
+             * make it easier to determine when a config file should be loaded.
+             */
+
             const basePath = path.resolve(cwd, globParent(pattern));
 
-            // group in cwd if possible and split out others
-            if (isPathInside(basePath, cwd)) {
-                ({ patterns: globbyPatterns, rawPatterns } = searches.get(cwd));
-            } else {
-                if (!searches.has(basePath)) {
-                    searches.set(basePath, { patterns: [], rawPatterns: [] });
-                }
-                ({ patterns: globbyPatterns, rawPatterns } = searches.get(basePath));
+            if (!searches.has(basePath)) {
+                searches.set(basePath, { patterns: [], rawPatterns: [] });
             }
+            ({ patterns: globbyPatterns, rawPatterns } = searches.get(basePath));
```

Comment:
The same change should be done for directories (lines 585-593) as they work the same as globs.